### PR TITLE
Refactor: tddft code

### PR DESCRIPTION
**Pull request by lyb9812**
1. refactor code in module_tddft, split LCAO_evolve.cpp to several files
2. simplify esolver_tddft, delete useless codes
3. add force of electric field for tddft
4. move read_parameters() of electric field for tddft from H_TDDFT_pw.cpp to input_conv.cpp

### CODE REVIEW

File: source/module_elecstate/potentials/H_TDDFT_pw.h [hunk 1]
```diff
@@ -21,6 +21,8 @@ class H_TDDFT_pw : public PotBase
 
     void cal_fixed_v(double* vl_pseudo) override;
 
+    static void read_parameters(Input *in);
```
Comment (dyzheng) [inline source/module_elecstate/potentials/H_TDDFT_pw.h]: Annotations with doxygen format for new functions please.

File: source/module_hamilt_lcao/module_tddft/bandenergy.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
+#include "bandenergy.h"
```
Comment (dyzheng) [inline source/module_hamilt_lcao/module_tddft/bandenergy.cpp:1]: In these files, git history can not find the real history of these code. you can choose some methods:
1. add annotations for these files, list which file and which commit can find the history code.
2. add UTs for functions in these files.

**Review by dyzheng — CHANGES_REQUESTED** [review]

File: source/module_hamilt_lcao/module_tddft/bandenergy.cpp (hunk 2 above)
Comment (lyb9812) [reply source/module_hamilt_lcao/module_tddft/bandenergy.cpp:1]: Annotations are added already. UTs will be added in next PR.

**Comment by lyb9812** [comment]
Fix #1935

File: source/module_io/input_conv.h [hunk 3]
```diff
@@ -24,6 +25,23 @@ void Convert(void);
 // fn (string): expressions such as "3*1 0 2*0.5 3*0"
 // arr (vector): stores parsing results, for example, "3*1 0 2*0.5 1*1.5" can be parsed as [1, 1, 1, 0, 0.5, 0.5, 1.5]
 template <typename T> void parse_expression(const std::string &fn, std::vector<T> &arr);
+
+#ifdef __LCAO
+/**
+ * @brief convert units
```
Comment (dyzheng) [inline source/module_io/input_conv.h]: From which unit to which unit?

File: source/module_io/input_conv.h [hunk 4]
```diff
@@ -24,6 +25,23 @@ void Convert(void);
...
+ *
+ * @param params input parameter
+ * @param c coefficients of unit conversion
+ * @return parameter after unit vonversion
+ */
+std::vector<double> convert_units(std::string params, double c);
```
Comment (dyzheng) [inline source/module_io/input_conv.h:37]: UT for this function should be added.

File: source/module_io/input_conv.h [hunk 5]
```diff
@@ -24,6 +25,23 @@ void Convert(void);
...
+
+/**
+ * @brief read paramers of electric field for tddft and convert units
+ */
+void read_td_efield();
```
Comment (dyzheng) [inline source/module_io/input_conv.h:42]: UT for this function should be added

File: source/module_elecstate/potentials/H_TDDFT_pw.h [hunk 6]
```diff
@@ -21,80 +22,79 @@ class H_TDDFT_pw : public PotBase
...
     void cal_fixed_v(double* vl_pseudo) override;
 
-  private:
-    // internal time-step,
-    //-------hypothesis-------
-    // Vext will evolve by time, every time cal_fixed_v() is called, istep++
-    //------------------------
-    static int istep;
+    /**
+     * @brief compute force of electric field
```
Comment (dyzheng) [inline source/module_elecstate/potentials/H_TDDFT_pw.h:26]: GPT said "The function takes in the cell information of the unit cell and calculates the force of the electric field using the dipole moment and other parameters." which can not get from annotations, maybe the brief part is too simple.

**Review by dyzheng — CHANGES_REQUESTED** [review]
Annotations is almost okay but some unittests is lack.

**Comment by dyzheng** [comment]
please fix conflicts for this PR

File: source/module_elecstate/potentials/H_TDDFT_pw.h (hunk 6 above)
Comment (lyb9812) [reply source/module_elecstate/potentials/H_TDDFT_pw.h:26]: It maybe mislead by the name fdip. I change the name and add formula for this force.

File: source/module_io/input_conv.h (hunk 3 above)
Comment (lyb9812) [reply source/module_io/input_conv.h]: I add description in annotation.

**Review by dyzheng — APPROVED** [review]

**Review by dyzheng — APPROVED** [review]

**Pull request merged**